src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -1606,8 +1606,21 @@ ntp_archiver::make_segment_index(
   retry_chain_logger& ctxlog,
   std::string_view index_path,
   ss::input_stream<char> stream) {
-    auto base_kafka_offset = model::offset_cast(
-      _parent.log()->from_log_offset(base_rp_offset));
+    std::exception_ptr eptr;
+    auto base_kafka_offset = [this, base_rp_offset, &eptr]() -> kafka::offset {
+        try {
+            return model::offset_cast(
+              _parent.log()->from_log_offset(base_rp_offset));
+        } catch (const std::runtime_error&) {
```

Comment:
maybe we should `catch (...)` so we aren't sensitive to changes in the exception type used by the offset translator?